### PR TITLE
RIA-8969 add_progress_migrated_case

### PR DESCRIPTION
**Pull request by ikirsanov**
### Jira link (if applicable)
https://tools.hmcts.net/jira/browse/RIA-8969

### Change description ###
Created new event for moving Migrated cases to desired state handler and junit tests.

### Checklist

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/AriaCreateCaseHandler.java [hunk 1]
```diff
@@ -48,7 +48,7 @@ public boolean canHandle(
         requireNonNull(callback, "callback must not be null");
 
         return callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT
-            && Event.ARIA_CREATE_CASE == callback.getEvent();
+            && (Event.ARIA_CREATE_CASE == callback.getEvent() || Event.START_APPEAL == callback.getEvent());
```
Comment (ThomasKKC) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/AriaCreateCaseHandler.java:51]: Also check if isAriaMigrated is Yes for START_APPEAL please.

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/migratedcase/ProgressMigratedCaseHandler.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.State;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.Callback;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.PreSubmitCallbackStage;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.field.YesOrNo;
+import uk.gov.hmcts.reform.iacaseapi.domain.handlers.PreSubmitCallbackHandler;
+
+
+@Component
+public class ProgressMigratedCaseHandler implements PreSubmitCallbackHandler<AsylumCase> {
```
Comment (ThomasKKC) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/migratedcase/ProgressMigratedCaseHandler.java:21]: Can we add FT for this please

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/entities/ccd/State.java [hunk 3]
```diff
@@ -5,52 +5,58 @@
 
 public enum State {
```
Comment (ThomasKKC) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/entities/ccd/State.java:6]: Can we update unit tests for this class?

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/entities/ccd/Event.java [hunk 4]
```diff
@@ -140,6 +140,7 @@ public enum Event {
     UPDATE_TRIBUNAL_DECISION("updateTribunalDecision"),
     MARK_APPEAL_AS_REMITTED("markAppealAsRemitted"),
     ARIA_CREATE_CASE("ariaCreateCase"),
+    PROGRESS_MIGRATED_CASE("progressMigratedCase"),
```
Comment (ThomasKKC) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/entities/ccd/Event.java:143]: Do we not need to add this event for adminofficer in application.yaml?

**Review by ThomasKKC — CHANGES_REQUESTED** [review]

**Review by ThomasKKC — APPROVED** [review]

**Review by MiguelMartinRojas — APPROVED** [review]

**Pull request merged**